### ISSUE TITLE
Convert to typescript

### ISSUE THREAD

**Issue by vraravam** [ferdium/ferdium-app#42]
### Your issue

The conversion to typescript was started by @mcmxcdev and myself back in the ferdi days. That effort was done piece-meal in small chunks, but has since come to a state where some portion of the codebase is in `ts` while other portions are in `js`. Need to restart that conversion

**Comment by SlanDF** [comment]
@SpecialAro is this issue ticket still relevant?

**Comment by SpecialAro** [comment]
> @SpecialAro is this issue ticket still relevant?

Yes. It's sill an ongoing task.

**Comment by cino** [comment]
@SlanDF 
![image](https://user-images.githubusercontent.com/2735602/178804742-77f039e7-12dd-4294-b4a0-a094237a8a59.png)

Highly relevant as you can see :)

**Comment by balajiv113** [comment]
I am looking to contribute towards this, is there any priority on which needs to be done first ??

**Comment by cino** [comment]
@balajiv113 There isn't really any priority on it, I'd suggest anything can be converted, I would just ignore the API folder as there is another pull request open with a [major upgrade ](https://github.com/ferdium/ferdium-server/pull/47) of the whole API which will be ported/integrated here.

**Comment by ghost** [comment]
Hi Team,
FYI - After the commit #797 merge, we shall reached a stage where all Ferdium app files ( excluding internal-server files) are converted to TS

Regarding internal-server approx. ~38 files (8.9% approx.) are in js form 
which shall be removed after implementation of V2 TS ferdium-server - plan is to use same ferdium server(configured for single tenant) here as submodule in ferdium app.

PFB - TS conversion progress.

![image](https://user-images.githubusercontent.com/104364298/203855046-5f8ecfe8-3d25-42eb-9afa-95ea1115494a.png)

**Comment by ghost** [comment]
FYI -  I shall further involve in cleanup activities that are spilled over in transformation


Queue:
- [ ] need to update eslint settings w.r.t typescript update and try to remove other quality check tools like prettier or common lint or editor config (After discussion and validation with team)
- [ ] refactor all ui components.
- [ ] pass readOnly to the display strings as Input (to avoid user from seeing cursor)
- [ ] remove all ()=>{} to noop (import from lodash) for cleaner
- [ ] check/ correct typing, style of code and refactor if needed
- [ ] Remove all any (return type)
- [ ] try to implement  private variable if needed
- [ ] Handle import cycles
- [ ] Clear all tech dept occured in this TS transformation (address TODO)

Thanks for support folks :)
